SConscript

```diff
@@ -464,7 +464,7 @@ def buildLibs(projectDir):
         '00std': [                                                              ],
         '01bsp': [                                                      'libbsp'],
         '02drv': [                                         'libdrivers','libbsp'],
-        '03oos': ['libopenstack','libopenapps','libkernel','libdrivers','libbsp'], # this order needed for mspgcc
```

Comment:
If we leave libdrivers compilation before libpsp, we get:

scons.bat board=OpenMote-CC2538 toolchain=armgcc apps=csensors oos_openwsn 
scons: Reading SConscript files ...

 ___                 _ _ _  ___  _ _ 
| . | ___  ___ ._ _ | | | |/ __>| \ |
| | || . \/ ._>| ' || | | |\__ \|   |
`___'|  _/\___.|_|_||__/_/ <___/|_\_|
     |_|                  openwsn.org

scons: done reading SConscript files.
scons: Building targets ...
Compiling projects\common\03oos_openwsn\03oos_openwsn.o
Compiling build\OpenMote-CC2538_armgcc\openstack\openstack.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02a-MAClow\topology.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02a-MAClow\IEEE802154.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02a-MAClow\IEEE802154E.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02a-MAClow\adaptive_sync.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02b-MAChigh\neighbors.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02b-MAChigh\otf.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02b-MAChigh\processIE.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02b-MAChigh\schedule.o
Compiling build\OpenMote-CC2538_armgcc\openstack\02b-MAChigh\sixtop.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03a-IPHC\iphc.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03a-IPHC\openbridge.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03b-IPv6\forwarding.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03b-IPv6\icmpv6.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03b-IPv6\icmpv6echo.o
Compiling build\OpenMote-CC2538_armgcc\openstack\03b-IPv6\icmpv6rpl.o
Compiling build\OpenMote-CC2538_armgcc\openstack\04-TRAN\opencoap.o
Compiling build\OpenMote-CC2538_armgcc\openstack\04-TRAN\opentcp.o
Compiling build\OpenMote-CC2538_armgcc\openstack\04-TRAN\openudp.o
Compiling build\OpenMote-CC2538_armgcc\openstack\cross-layers\idmanager.o
Compiling build\OpenMote-CC2538_armgcc\openstack\cross-layers\openqueue.o
Compiling build\OpenMote-CC2538_armgcc\openstack\cross-layers\openrandom.o
Compiling build\OpenMote-CC2538_armgcc\openstack\cross-layers\packetfunctions.o
Archiving build\OpenMote-CC2538_armgcc\openstack\libopenstack.a
Indexing  build\OpenMote-CC2538_armgcc\openstack\libopenstack.a
Dynifying build\OpenMote-CC2538_armgcc\openapps\openapps_dyn.c
Compiling build\OpenMote-CC2538_armgcc\openapps\openapps_dyn.o
Compiling build\OpenMote-CC2538_armgcc\openapps\c6t\c6t.o
Compiling build\OpenMote-CC2538_armgcc\openapps\cinfo\cinfo.o
Compiling build\OpenMote-CC2538_armgcc\openapps\cleds\cleds.o
Compiling build\OpenMote-CC2538_armgcc\openapps\cwellknown\cwellknown.o
Compiling build\OpenMote-CC2538_armgcc\openapps\rrt\rrt.o
Compiling build\OpenMote-CC2538_armgcc\openapps\techo\techo.o
Compiling build\OpenMote-CC2538_armgcc\openapps\uecho\uecho.o
Archiving build\OpenMote-CC2538_armgcc\openapps\libopenapps.a
Indexing  build\OpenMote-CC2538_armgcc\openapps\libopenapps.a
Compiling build\OpenMote-CC2538_armgcc\kernel\openos\scheduler.o
Archiving build\OpenMote-CC2538_armgcc\kernel\openos\libkernel.a
Indexing  build\OpenMote-CC2538_armgcc\kernel\openos\libkernel.a
Compiling build\OpenMote-CC2538_armgcc\drivers\common\openhdlc.o
Compiling build\OpenMote-CC2538_armgcc\drivers\common\opensensors.o
Compiling build\OpenMote-CC2538_armgcc\drivers\common\openserial.o
Compiling build\OpenMote-CC2538_armgcc\drivers\common\opentimers.o
Compiling build\OpenMote-CC2538_armgcc\drivers\OpenMote-CC2538\adxl346.o
Compiling build\OpenMote-CC2538_armgcc\drivers\OpenMote-CC2538\max44009.o
Compiling build\OpenMote-CC2538_armgcc\drivers\OpenMote-CC2538\sht21.o
Archiving build\OpenMote-CC2538_armgcc\drivers\libdrivers.a
Indexing  build\OpenMote-CC2538_armgcc\drivers\libdrivers.a
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\adc_sensor.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\board.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\bsp_timer.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\debugpins.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\eui64.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\i2c.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\leds.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\radio.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\radiotimer.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\sensors.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\startup_gcc.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\uart.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\adc.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\aes.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\ccm.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\cpu.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\debug.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\flash.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\gpio.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\gptimer.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\i2c.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\interrupt.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\ioc.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\pka.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\sha256.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\sleepmode.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\ssi.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\sys_ctrl.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\systick.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\uarthal.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\udma.o
Compiling build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\source\watchdog.o
Archiving build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\libbsp.a
Indexing  build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\libbsp.a
Linking   projects\common\03oos_openwsn_prog.exe
build\OpenMote-CC2538_armgcc\bsp\boards\OpenMote-CC2538\libbsp.a(sensors.o): In function `sensors_init':
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:11: undefined reference to `sht21_is_present'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:12: undefined reference to `sht21_init'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:24: undefined reference to `max44009_is_present'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:25: undefined reference to `max44009_init'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:32: undefined reference to `adxl346_is_present'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:33: undefined reference to `adxl346_init'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `sht21_read_temperature'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `sht21_convert_temperature'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `sht21_read_humidity'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `sht21_convert_humidity'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `max44009_read_light'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `max44009_convert_light'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `adxl346_read_x'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `adxl346_read_y'
C:\Users\Nicola\Desktop\openwsn-current\openwsn-fw/bsp\boards\OpenMote-CC2538/sensors.c:58: undefined reference to `adxl346_read_z'
collect2.exe: error: ld returned 1 exit status
scons: *** [projects\common\03oos_openwsn_prog.exe] Error 1
scons: building terminated because of errors.

Changing to compile libdrivers after libbsp works ok, in fact Travis tells that this is ok for all platforms.